src/Assets/PaymentMethodAssets.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Handles assets for registered payment types.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\Assets;
+
+use Automattic\WooCommerce\Blocks\Payments\PaymentMethodRegistry;
+
+/**
+ * PaymentMethodAssets class.
+ *
+ * @since $VID:$
+ */
+class PaymentMethodAssets {
+
+	/**
+	 * Reference to the PaymentMethodRegistry instance
+	 *
+	 * @var PaymentMethodRegistry
+	 */
+	private $payment_method_registry;
+
+	/**
+	 * An instance of the AssetDataRegistry
+	 *
+	 * @var AssetDataRegistry
+	 */
+	private $asset_registry;
+
+	/**
+	 * Constructor for class
+	 *
+	 * @param PaymentMethodRegistry $payment_method_registry An instance of Payment Method Registry.
+	 * @param AssetDataRegistry     $asset_registry  Used for registering data to pass along to the request.
+	 */
+	public function __construct( PaymentMethodRegistry $payment_method_registry, AssetDataRegistry $asset_registry ) {
+		$this->payment_method_registry = $payment_method_registry;
+		$this->asset_registry          = $asset_registry;
+	}
+
+	/**
+	 * Gets an array of all registered payment method script handles.
+	 *
+	 * @return string[]
+	 */
+	public function get_all_registered_payment_method_script_handles() {
+		$script_handles  = [];
+		$payment_methods = $this->payment_method_registry->get_all_registered();
+
+		foreach ( $payment_methods as $payment_method ) {
+			$script_handles = array_merge(
+				$script_handles,
+				is_admin() ? $payment_method->get_payment_method_script_handles_for_admin() : $payment_method->get_payment_method_script_handles()
+			);
+		}
+
+		return array_unique( array_filter( $script_handles ) );
```

Comment:
Good call on simplifying things here . I get why `array_unique`, but why `array_filter`?  Is it used to guard against non truthy array values that might have been provided by the payment method?

Reply:
yeah; this is a shortcut to remove empty values too.